### PR TITLE
Eaton M2 EMP g2

### PR DESCRIPTION
**Pull request by dagbdagb**
The Eaton M2 NMC for current generation UPSes has a companion Environmental Monitoring Probe gen 2. (temp, humidity, 2x contacts). This adds minimal support for said probe. I choose to use the (simpler) xups mib over the much more detailed eaton-sensor-mib.

This PR also has an updated snmprec for a device with this probe, as well as with an updated M2 firmware version (2.0.5)

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Hi,

Please see failed tests

**Comment by dagbdagb** [comment]
The failed tests reference stuff I haven't touched. 
Additionally, I have added new json- and snmprec-files, as variants. (New major firmware release + new external sensor. Unclear if these two points warrants a variant.) 

One thing I observe after my additions is that my new sensors have weird types under 'Health settings'. The two contacts are type xupsEnvironment, while temp/hum are of type eatonupsm2. Not sure if this implies a bug or a need for adjustments.

Would very much appreciate some handholding here. As time and energy permits, of course.

**Comment by PipoCanaja** [comment]
> The failed tests reference stuff I haven't touched.
As soon as you touch an existing YAML OS file, you will impact any device that uses it. This is why we have regression tests. If the resulting change is acceptable, then the regression tests will be changed. If they are not acceptable, then you'll need to adapt your patch. 
So please run save-test-data on every failing regression test file (as seen in the test results).

> Additionally, I have added new json- and snmprec-files, as variants. (New major firmware release + new external sensor. Unclear if these two points warrants a variant.)
This is good practice (and required) to add new test data for every change, to ensure regression tests will also check your new addition.

File: includes/definitions/discovery/eatonupsm2.yaml [hunk 1]
```diff
@@ -13,8 +13,26 @@ modules:
             data:
                 -
                     oid: xupsEnvironment
-                    value: xupsEnvAmbientTemp
```
Comment (PipoCanaja) [inline includes/definitions/discovery/eatonupsm2.yaml:16]: Hi @dagbdagb 
Why did you remove this sensor. Are you sure it cannot be of any use for anybody ? Removing it here will remove it for each and every LibreNMS user after this PR is merged. In doubt, I would keep it, and also add you xupsEnvRemoteTemp,  so you end up with both.

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

File: includes/definitions/discovery/eatonupsm2.yaml (hunk 1 above)
Comment (dagbdagb) [reply includes/definitions/discovery/eatonupsm2.yaml:16]: > Hi @dagbdagb
> Why did you remove this sensor. Are you sure it cannot be of any use for anybody ? Removing it here will remove it for each and every LibreNMS user after this PR is merged. In doubt, I would keep it, and also add you xupsEnvRemoteTemp, so you end up with both.

You are right of course. The current major release does not expose the onboard temperature sensor. But the previous release did, so we need to keep this. Improved version coming up. 

Thank you for reviewing.

**Review by PipoCanaja — APPROVED** [review]
LGTM. Thanx for your contribution
